packages/core/src/VNodes/CharNode.ts

```diff
@@ -1,10 +1,67 @@
 import { VNode, VNodeType, FormatType } from './VNode';
 
+/**
+ * This "phantom type" is there to ensure that the type `Char` is only generated
+ * through the use of the function `makeChar`, so as to force going through the
+ * length check.
+ */
+type InternalChar<T> = { valid: true } & string;
+export type Char = InternalChar<{}>;
+/**
+ * Return a Char type from a string of length 1 (validating the type).
+ *
+ * @param char
+ */
+export function makeChar(char: string): Char {
+    if (char.length === 1) {
+        return char as Char;
+    } else {
+        throw new Error('Cannot make a Char out of anything else than a string of length 1.');
+    }
+}
+
 export class CharNode extends VNode {
     properties = {
         atomic: true,
     };
-    constructor(char: string, format: FormatType) {
-        super(VNodeType.CHAR, '#text', char, format);
+    char: Char;
+    constructor(char: string, format?: FormatType) {
+        super(VNodeType.CHAR, '#text', format);
+        this.char = makeChar(char);
+        this.name = char;
+    }
+
+    //--------------------------------------------------------------------------
+    // Lifecycle
+    //--------------------------------------------------------------------------
+
+    /**
+     * Return a new VNode with the same type and attributes as this VNode.
+     *
+     * @override
+     */
+    shallowDuplicate(): CharNode {
+        return new CharNode(this.char, this.format);
+    }
+
+    //--------------------------------------------------------------------------
+    // Public
+    //--------------------------------------------------------------------------
+
+    /**
+     * Return the length of this VNode.
+     */
+    get length(): number {
```

Comment:
Why do we need this? Its length can be inferred from its type.

Reply:
_Because_ its length can be inferred from its type. This is an override of `VNode`'s `length` method, because in the case of a `CharNode` it shouldn't go look for how many children it has, but it's always 1. That said, I _should_ have put `@override` in the doc string.